source/common/network/listen_socket_impl.cc

```diff
@@ -37,6 +38,22 @@ TcpListenSocket::TcpListenSocket(const Address::InstanceConstSharedPtr& address,
   int rc = setsockopt(fd_, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on));
   RELEASE_ASSERT(rc != -1);
 
+#if defined SOL_IP && defined IP_TRANSPARENT
+  // Set the SOL_IP/IP_TRANSPARENT socket option to support running with iptables TPROXY target
+  // instead of REDIRECT to get both the original source and destination addresses of the
+  // redirected socket.
+  on = 1;
+  rc = setsockopt(fd_, SOL_IP, IP_TRANSPARENT, &on, sizeof(on));
+  if (rc == -1) {
+    if (errno == EPERM) {
+      // Ignore. Not running with CAP_NET_ADMIN capability.
```

Comment:
This is still a real error, if configuration requires `IP_TRANSPARENT`.

Reply:
Fixed in upcoming commit.